### PR TITLE
docs: add a section about `useRequestFetch` and `event.$fetch`

### LINKED ISSUE #19815: document `event.$fetch`/`useRequestFetch`
follow up from https://github.com/nuxt/nuxt/pull/19357

### PR DESCRIPTION
**Pull request by cernymatej**
### 🔗 Linked issue

resolves https://github.com/nuxt/nuxt/issues/19815

### 📚 Description

This PR adds brief documentation for `useRequestFetch` and `event.$fetch`

### CODE REVIEW

File: docs/3.api/2.composables/use-request-fetch.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+The [`useFetch`](/docs/api/composables/use-fetch) composable uses `useRequestFetch` under the hood to automatically forward the request context and headers.
+::
+
+::code-group
+
+```vue [pages/index.vue]
+<script setup lang="ts">
+  // This will forward the user's headers to the `/api/foo` event handler
+  // Result: { cookies: { foo: 'bar' } }
+  const { data: forwarded } = await useAsyncData(() => useRequestFetch()('/api/cookies'))
```
Comment (danielroe) [inline docs/3.api/2.composables/use-request-fetch.md:31]:
```suggestion
  const requestFetch = useRequestFetch()
  const { data: forwarded } = await useAsyncData(() => requestFetch('/api/cookies'))
```

**Review by danielroe — APPROVED** [review]
This is fantastic - thank you 🙏

**Comment by manniL** [comment]
@CernyMatej Thanks for the contribution 🔥

I'm curious: did you watch [the video](https://www.youtube.com/watch?v=7vfwYaHtv3U) with the "call to contribution before ☺️

**Comment by CernyMatej** [comment]
@manniL Yes I did 😄
Great video as always btw

**Comment by atinux** [comment]
Could we cherry pick this commit to `3.x` somehow @danielroe ?

**Comment by danielroe** [comment]
thanks for spotting!

**Pull request merged**